src/sprout/Helpers/FileTransform.php

```diff
@@ -0,0 +1,578 @@
+     *
+     * @param string $transform_filename
+     *
+     * @return FileTransformModel|null
+     */
+    public static function getByTransformFilename(string $transform_filename)
+    {
+        try {
+            return FileTransformModel::findOne(['transform_filename' => $transform_filename]);
+        } catch (RowMissingException $e) {
+        }
+
+        return null;
+    }
+
+
+    /**
+     * Look for a transform table entry for a file by the filename
+     *
+     * If not found, this will attempt to find matching file details in the files table
+     * If we have these, we'll try and ID lookup
+     *
+     * @param string $filename The name of the file (no path) we're looking for
+     * @param string $transform_name The transform string e.g. 'c400x200'
+     *
+     * @return FileTransformModel|null
+     */
+    public static function findByFilename(string $filename, string $transform_name): ?FileTransformModel
+    {
+        try {
+            return FileTransformModel::findOne(['filename' => $filename, 'transform_name' => $transform_name]);
+        } catch (RowMissingException $e) {
+        }
+
+        return null;
+    }
+
+
+    /**
+     * Look for a transform table entry for a file by its ID
+     *
+     * @param int $file_id The ID of the file we're looking for
+     * @param string $transform_name The filesize string e.g. 'c400x200'
+     *
+     * @return FileTransformModel|null
+     */
+    public static function findByFileId(int $file_id, string $transform_name): ?FileTransformModel
+    {
+        try {
+            return FileTransformModel::findOne(['file_id' => $file_id, 'transform_name' => $transform_name]);
+        } catch (RowMissingException $e) {
+        }
+
+        return null;
+    }
+
+
+    /**
+     * Create default image sizes as per the config parameter 'file.image_transformations'
+     *
+     * @param int $file_id The file to create sizes for
+     *
+     * @return bool
+     */
+    public static function createDefaultTransforms(int $file_id)
+    {
+        $sizes = Kohana::config('file.image_transformations');
+        return FileTransform::createTransformSizes($file_id, $sizes);
+    }
+
+
+    /**
+     * Create default image sizes as per the config parameter 'file.image_transformations_instant'
+     *
+     * @param int $file_id The file to create sizes for
+     *
+     * @return bool
+     */
+    public static function createInstantTransforms(int $file_id)
+    {
+        $sizes = Kohana::config('file.image_transformations_instant');
+        return FileTransform::createTransformSizes($file_id, $sizes);
+    }
+
+
+    /**
+     * Create default image sizes as per the specified array of sizes'
+     *
+     * @see config file.image_transformations*
+     *
+     * The transformed files get saved onto the server.
+     * If any of the transformations in a transform-group fails,
+     * the whole group will fail and the file will not be saved.
+     *
+     * @throw Exception
```

Comment:
This a good opportunity to use a specific exception type so consumers can differentiates between deeply thrown errors. For when this is inevitably wrapped 5 times over in some other helper.

We already have ImageException that might fit.

Reply:
Sounds like a job for... `FileTransformException` 

Honestly I'm all for exception specificity these days, and I'd prefer a folder with 30 classes in it, than trying to imply meaning from a message or code in a more generic error